src/extension.c

```diff
@@ -74,11 +74,18 @@ ts_extension_check_version(const char *so_version)
 		 */
 		ereport(FATAL,
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-				 errmsg("extension \"%s\" version mismatch: shared library version %s; SQL version "
+				 errmsg("extension \"%s\" version mismatch: shared library version %s; client "
+						"version "
 						"%s",
 						EXTENSION_NAME,
 						so_version,
-						sql_version)));
+						sql_version),
+				 errdetail("A client submitted a SQL request to TimescaleDB using different "
+						   "version of the extension. "
```

Comment:
I don't understand this. Are SQL requests tagged with version of the extension they are intended for?